### PR TITLE
Fixes #426 Dropped class HandyReturnValues 

### LINKED ISSUE #426: Refactor instance based utility classes to static utility classes
The internal implementation of Mockito contains numerous helper classed that serve as a container for useful methods. By its nature these classes don't have a state. Currently some them are instance based in other word you need to create the utility class to call a helper method. This not only pollutes the heap but also the code cause instance methods can't be imported statically. 

Here is an example:

```java
Helper helper = new Helper();
if (helper.isInputValid(input)){
  [...] 
}
```

vs. static import of Helper.isInputValid

```java
if (isInputValid(input)){
  [...] 
}
```

The aim of this ticket is to identify canidates that can be refactored to static utility classes. If you like to refactoring and mockito feel free to send a PR and reference this issue.

Refactoring canidates:
- [ ] `AccessibilityChanger`
- [ ] `BeanPropertySetter`
- [ ] `ConditionalStackTraceFilter`
- [ ] `FieldCopier`
- [ ] `FieldReader`
- [ ] `GenericMaster` should be integrate into `GenericTypeResolver`
- [ ] `JUnitFailureHacker` can be removed when the deprecated `VerboseMockitoJUnitRunner` is removed
- [ ] `LenientCopyTool`
- [ ] `MatcherBinder`
- [ ] `MockitoCore` should better be a singleton
- [ ] `MockCreationValidator`
- [ ] `RemoveFirstLine`
- [x] #591 `ArgumentMatchingTool`
- [x] #515 `AllInvocationsFinder`
- [x] #502 `ArgumentsComparator`
- [x] #540 `ArrayUtils`
- [x] #490 `AtLeastXNumberOfInvocationsChecker`
- [x] #490 `AtLeastXNumberOfInvocationsInOrderChecker`
- [ ] #912 `Constructors`
- [x] #427 `FieldSetter`
- [x] #908 `FriendlyExceptionMaker`
- [x] #431 `HandyReturnValues` 
- [x] #432 `InvocationMarker`
- [x] #462 `InvocationsFinder`
- [x] #908 `JUnitDetecter`
- [x] #490 `MissingInvocationChecker`
- [x] #490 `MissingInvocationInOrderChecker`
- [x] #514 `MockUtil`
- [x] #503 `NonGreedyNumberOfInvocationsInOrderChecker`
- [x] #907 `NumberOfInvocationsInOrderChecker`
- [x] #907 `NumberOfInvocationsChecker`
- [x] #547 `ObjectMethodsGuru`
- [x] #427 `Reporter`
- [x] #535 `SuperTypesLastSorter`
- [x] #501 `TestMethodFinder`
- [x] #515 `VerifiableInvocationsFinder`

### PR DESCRIPTION
**Pull request by ChristianSchwarz**
While refactoring the class` HandyReturnValues` towards a static utility class I notices that it is not required at all. Therefore I decided to remove the type `HandyReturnValues`.

### CODE REVIEW

File: src/main/java/org/mockito/AdditionalMatchers.java [hunk 1]
```diff
@@ -35,958 +36,1027 @@
  *   
  *   //1 or 10
  *   mock.someMethod(or(eq(1), eq(10)));
- * </code></pre>
+ * </code>
+ * </pre>
  * 
  * Scroll down to see all methods - full list of matchers.
  */
 public class AdditionalMatchers {
```
Comment (TimvdLippe) [inline src/main/java/org/mockito/AdditionalMatchers.java]: Please disable your autoformatter and revert the changes in this class.

**Comment by TimvdLippe** [comment]
Hm so apparently it is not just that class. Could you revert all formatting changes and only commit the actual removed class + usages? That way this is reviewable (and mergable, as it results in a lot of merge conflicts).

**Comment by ChristianSchwarz** [comment]
@TimvdLippe Yes I know, I am going to fix it....

**Comment by szpak** [comment]
I use `HandyReturnValues` in [mockito-java8](https://github.com/szpak/mockito-java8/blob/22c0d937c326f2f8938b6e10721c4ed31a1015c1/src/main/java/info/solidsoft/mockito/java8/LambdaAwareHandyReturnValues.java). It will break compatibility, but if `Primitives.defaultValue(clazz)` was a drop-in replacement (@ChristianSchwarz ?) I could switch to it in the next version.

**Comment by ChristianSchwarz** [comment]
#431 replaces this PR

**Comment by ChristianSchwarz** [comment]
@szpak `HandyReturnValues `was in an internal package, it should not break compatibility. If you need a default value for primitives or wrapper types you can use the method.

**Comment by szpak** [comment]
The current mockito-java8 version will not work with the Mockito version after that change. However, as I mentioned it's not a big issue (it's still beta).
